src/parser/mod.rs

```diff
@@ -15017,6 +15026,48 @@ impl<'a> Parser<'a> {
         }
     }
 
+    fn parse_go(&mut self) -> Result<Statement, ParserError> {
```

Comment:
One thing we could do, in the `parse_stmt()` function could we call `self.prev_token()` to rewind the `Go` keyword so that this function becomes self contained in being able to parse a full `Go` statement?